client/lib/feed-stream-store/paged-stream.js

```diff
@@ -248,11 +244,23 @@ assign( PagedStream.prototype, {
 			} );
 			this.postKeys = this.postKeys.concat( postKeys );
 			this.page++;
-			this.emit( 'change' );
+			this.emitChange();
 		}
 	}
-} );
 
-Emitter( PagedStream.prototype );
+	dismissPostsFromSite( blogId ) {
+		const removed = remove( this.postKeys, { blogId } );
+		if ( removed.length === 0 ) {
+			return; // nothing changed
+		}
+		// reshuffle what we have
+		this.postKeys = shuffle( this.postKeys );
+		// remove from the keyed index
+		forEach( removed, ( { postId } ) => {
```

Comment:
switched to using a `Set`. We never used the value and setting it to `false` here was actually broken for a check elsewhere. Plus `Set` should be faster that using an object-as-a-map on most browsers.